File: rabbitmq-sample/src/main/java/yun/blog/rabbitmqsample/member/MemberApi.java
package yun.blog.rabbitmqsample.member;

import javax.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/members")
@RequiredArgsConstructor
public class MemberApi {

  private final MemberSignUpService memberSignUpService;

  @PostMapping
  public Member doSignUp(@RequestBody SignUpRequest dto){
    return memberSignUpService.doSignUp(dto);
  }
}

File: kotlin-junit5/src/main/kotlin/com/example/kotlinjunit5/member/MemberApi.kt
package com.example.kotlinjunit5.member

import org.springframework.web.bind.annotation.*
import javax.validation.constraints.NotEmpty

@RestController
@RequestMapping("/members")
class MemberApi(
        val memberRepository: MemberRepository
) {

    @PostMapping
    fun create(@RequestBody dto: MemberSignUpRequest) {
        memberRepository.save(dto.toEntity())
    }

    @GetMapping
    fun getMembers(): List<Member> {
        return memberRepository.findAll()
    }
}

data class MemberSignUpRequest(
        @field:NotEmpty
        val name: String,
        @field:NotEmpty
        val email: String
) {

    fun toEntity(): Member {
        return Member(email, name)
    }
}